BUG: make special.eval_hermite return nan if second argument is nan

peytondmurray opened the pull request:
#### Reference issue
Closes #11369.

#### What does this implement/fix?
* The `special.eval_hermitenorm` and `special.eval_hermite` (which calls `eval_hermitenorm`) functions save a little computation work for the special case where the first argument is `0`, in which case the return value is always `1`; however, in the case where the second argument is `np.nan`, we'd like the `np.nan` value to propagate through. The modifications made here ensure both functions correctly return `np.nan` when the second argument passed to the function is `np.nan`.
* Tests have been added for both functions to check that `np.nan` is correctly returned when passed as the second argument.

peytondmurray commented:
@pvanmulbregt I plan to address #11370, #11361, and several other functions in `scipy.special` as well, all of which appear to be dropping `np.nan` values. Would it be easiest if I address them all here in a single PR, or would you prefer if I made separate PRs for each?

pvanmulbregt commented:
That question may be best answered by @person142, who may have spent the most time recently thinking about `scipy.special`.   As well as whether the return should be `np.nan`, 
or whether it should be the same nan that was passed in (preserving the type of nan).

person142 commented:
Separate PRs please. And yes, do propagate the NaN instead of making a new one, we e.g want to make sure an sNaN is propagated if at all possible.

peytondmurray commented:
@person142 Thanks for the review. The second parameter is now returned if it is `np.nan`, preserving the type. I've squashed and rebased onto master, and ensured that local tests are passing.

person142 commented on scipy/special/tests/test_orthogonal_eval.py:
```diff
@@ -252,3 +253,11 @@ def test_hermite_domain():
+
+@pytest.mark.parametrize("n", [0, 1, 2])
+@pytest.mark.parametrize("x", [0, 1, np.nan])
+def test_hermite_nan(n, x):
```
One more tiny thing: either name the test something like `test_gh_11369` or link to the issue in a comment so we know what it’s a regression test for.

Also we usually don’t put docstrings on tests because they make the pytest outputs too verbose.

peytondmurray replied:
:+1: Removed the docstring, replaced with reference to the original issue using the same format as in the test above.

person142 commented:
Thanks @peytondmurray.
